### PR TITLE
[Improvement](decimal) reduce overhead on disable check decimal overflow

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes

reduce overhead on disable check decimal overflow
tpchq1 11s -> 9s

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

File: be/src/vec/data_types/number_traits.h [hunk 1]
```diff
@@ -220,7 +221,6 @@ template <typename T>
 /// Returns the maximum ascii string length for this type.
 /// e.g. the max/min int8_t has 3 characters.
 int max_ascii_len() {
-    LOG(FATAL) << "Not implemented.";
```
Comment (HappenLee) [inline be/src/vec/data_types/number_traits.h:223]: why not add dcheck or exception?

Comment (BiteTheDDDDt) [reply be/src/vec/data_types/number_traits.h:223]: On be/src/vec/functions/function_cast.h line 280, I use max_ascii_len to judge multiply may overflow

File: be/src/vec/data_types/data_type_decimal.h [hunk 2]
```diff
@@ -410,73 +412,69 @@ constexpr bool IsDataTypeDecimalOrNumber =
...
+    using FromFieldType = typename FromDataType::FieldType;
+    using ToFieldType = typename ToDataType::FieldType;
+    using MaxFieldType = std::conditional_t<(sizeof(FromFieldType) > sizeof(ToFieldType)),
+                                            FromFieldType, ToFieldType>;
 
-    return converted_value;
+    // from decimal to integer
+    MaxFieldType multiplier = DataTypeDecimal<MaxFieldType>::get_scale_multiplier(scale_from);
+    for (size_t i = 0; i < size; i++) {
+        dst[i] = static_cast<MaxFieldType>(src[i]).value / multiplier.value;
```
Comment (jacktengg) [inline be/src/vec/data_types/data_type_decimal.h:472]: need to check overflow, e.g. cast decimal128 to int.

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by jacktengg — APPROVED** [review]

**Review by zhangstar333 — APPROVED** [review]

**Pull request merged**